### DISCUSSION TITLE
Nested Variables in a Variable

### DISCUSSION THREAD

**Discussion by bfrancom** [community/community#27207] [Actions]
<p>I am trying to figure out how to get this <code>NETLIFY_SITE_ID</code> variable working.</p>
<pre><code class="lang-auto">    frontend-build:
#      needs: build
      runs-on: ubuntu-18.04
      steps:
      - uses: actions/checkout@v2

    # Deploy the site with netlify-cli
      - name: Deploy Site
        env:
          GITHUB_SHA: ${{ steps.vars.outputs.github_short_sha }}
          NETLIFY_SITE_ID: ${{ format('{0}{1}{2}{3}{4}', '${{ ', 'secrets.', github.event.inputs.lane, '_NETLIFY_SITE_ID', ' }}') }}
          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
        run: |
          echo $NETLIFY_SITE_ID
          netlify deploy --dir=client/dist --build --prod --message "Deployed from GitHub Action sha $GITHUB_SHA ref ${{ steps.vars.outputs.short_ref }}"
</code></pre>
<p>The output in the console looks correct, but it’s not interpreting it as a secret:<br>
<img src="https://user-images.githubusercontent.com/5726729/181102241-8ac32a0e-4b17-40be-b10c-cdf10d1fda65.png" alt="netlify" data-base62-sha1="x9gKVIWVMnrREwPghe2KPzQASRV" width="481" height="287"></p>
<p>When it works, it should interpret it as a secret and mask the value. I can’t seem to figure it out.  Any help would be greatly appreciated. Thanks!</p>

**Comment by Simran-B** [comment, marked as answer]
<p>I think it needs to be:</p>
<pre><code class="lang-auto">NETLIFY_SITE_ID: ${{ secrets[format('{0}_NETLIFY_SITE_ID', github.event.inputs.lane)] }}
</code></pre>

**Reply by atulec016** [reply]
all the answers are from either the output from different step or input , can we do with the variable geneated within the same step

**Comment by bfrancom** [comment]
<p>Thank you! I was banging my head against a wall on this for so long. I really appreciate it!</p>
